### PR TITLE
Split up large writes in the EventSourcedRememberEntitiesShardStore

### PR DESCRIPTION
**Pull request by johanandren**

### CODE REVIEW

File: akka-cluster-sharding/src/main/resources/reference.conf [hunk 1]
```diff
@@ -148,6 +148,13 @@ akka.cluster.sharding {
     number-of-entities = 5
   }
 
+  event-sourced-remember-entities-store {
+    # When using remember entities and the event sourced remember entities store the batches
+    # written to the store are limited by this number to avoid getting a too large event for
+    # the journal to handle.
+    max-updates-per-write = 500
```
Comment (johanandren) [inline akka-cluster-sharding/src/main/resources/reference.conf:155]: I took this number out of thin air, any ideas what would be reasonable to not hit max atomic writes/limits in the most common journals?

**Comment by johanandren** [comment]
Unrelated failure

**Comment by johanandren** [comment]
PLS BUILD

File: akka-cluster-sharding/src/main/resources/reference.conf (hunk 1 above)
Comment (johanandren) [reply akka-cluster-sharding/src/main/resources/reference.conf:155]: @chbatey @patriknw

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberEntitiesShardStore.scala [hunk 2]
```diff
@@ -93,14 +96,17 @@ private[akka] final class EventSourcedRememberEntitiesShardStore(
...
+        .grouped(maxUpdatesPerWrite)
+        .foreach(group =>
+          persistAll(group) { _ =>
+            left -= 1
+            if (left == 0) {
+              sender() ! RememberEntitiesShardStore.UpdateDone(started, stopped)
+              state.copy(state.entities.union(started).diff(stopped))
+              saveSnapshotWhenNeeded()
+            }
+          })
```
Comment (johanandren) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberEntitiesShardStore.scala:109]: Could perhaps avoid some overhead when the write does not need splitting up.

File: akka-cluster-sharding/src/main/resources/reference.conf (hunk 1 above)
Comment (chbatey) [reply akka-cluster-sharding/src/main/resources/reference.conf:155]: Shall we go with 100 assuming a 50 byte key? That is a large key. The C* plugin will warn at a batch over 5kb and drop the write at a request over 50kb with the default settings

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberEntitiesShardStore.scala (hunk 2 above)
Comment (chbatey) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberEntitiesShardStore.scala:109]: 👍 which often will be the case

**Review by chbatey — APPROVED** [review]
Looks good. Up to you if you want to special case the single write case

**Review by patriknw — COMMENTED** [review]
looking good after the comments

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**